integration-tests/helpers.js

```diff
@@ -95,10 +95,13 @@ class FakeAgent extends EventEmitter {
 function spawnProc (filename, options = {}) {
   const proc = fork(filename, options)
   return new Promise((resolve, reject) => {
-    proc.on('message', ({ port }) => {
-      proc.url = `http://localhost:${port}`
-      resolve(proc)
-    }).on('error', reject)
+    proc
+      .on('message', ({ port }) => {
+        proc.url = `http://localhost:${port}`
+        resolve(proc)
+      })
+      .on('error', reject)
+      .on('exit', code => reject(new Error(`Process exited with status code ${code}.`)))
```

Comment:
Doesn't this mean we always reject if a process exits at all?

Reply:
Yes I forgot the condition, will add it now.